### PR TITLE
Feat: improve scalability and performance of OME-Zarr writer

### PR DESCRIPTION
**Pull request by carshadi**
This PR implements a blocked writing strategy for OME-Zarr datasets. Dask arrays are partitioned into non-overlapping blocks of size ~400GiB and processed serially. Parallel processing is done within each block. This basically eliminates the scheduler overhead and OOM errors I was seeing when processing whole tiles / brain volumes at once, and doesn't slow down the writing. For downsampling (diSPIM), the previous level is re-read from either the output s3 bucket (levels 1-n) or disk (level 0), which significantly speeds up pyramid generation since previously the full-resolution layer was re-read for each downsampling level. 

Changes
* `BlockedArrayWriter` class which handles the array partitioning and writing. `ome-zarr-py` is only used for writing the OME-NGFF metadata now.
* Drops support for Python `3.8`, necessary for latest versions of Dask
* Adds support for Python `3.11` - transcode job tested, ephys jobs *not tested* but all unit tests pass
* Updates dependencies `dask==2023.7.1`, `distributed==2023.7.1`, `ome-zarr-py==0.8.0`, `zarr==2.16.0`
* Fixes chunk computation methods
* Misc code cleanup

Tested on CentOS 7

### CODE REVIEW

**Comment by carshadi** [comment]
@jtyoung84 thoughts on requiring Python >= 3.9 ? Would be nice to be able to keep up with the latest Dask releases

**Comment by jtyoung84** [comment]
> @jtyoung84 thoughts on requiring Python >= 3.9 ? Would be nice to be able to keep up with the latest Dask releases

I think that should be okay.

File: src/aind_data_transfer/transformations/ome_zarr.py [hunk 1]
```diff
@@ -788,4 +781,203 @@ def _get_axes_5d(time_unit: str = "millisecond", space_unit: str = "micrometer")
...
+        The scale factors for downsampling along each dimension.
+    chunks : tuple
+        The chunk size to use for the output arrays.
+
+    Returns
+    -------
+    list
+        A list of Dask arrays representing the pyramid levels.
+    """
+    pyramid = multiscale(
```
Comment (camilolaiton) [inline src/aind_data_transfer/transformations/ome_zarr.py:812]: @carshadi, have you noticed aliasing in the downsampled resolutions? As far as I am concerned, we haven't checked this.

**Review by camilolaiton — APPROVED** [review]

File: src/aind_data_transfer/transformations/ome_zarr.py (hunk 1 above)
Comment (carshadi) [reply src/aind_data_transfer/transformations/ome_zarr.py:812]: I haven't noticed aliasing, but we could open a separate issue for that, unless it's related to these changes?

**Review by jwong-nd — APPROVED** [review]
BlockedArrayWriter has been working in the SOFIMA pipeline.

**Pull request merged**